taqueria-vscode-extension/src/lib/helpers.ts

```diff
@@ -391,19 +436,30 @@ export const inject = (deps: InjectedDependencies) => {
 		output: Output,
 		i18n: i18n,
 		projectDir: Util.PathToDir,
-		addConfigWatcherIfNotExists: (folder: string, factory: () => api.FileSystemWatcher) => void,
+		addConfigWatcherIfNotExists: (folder: string, factory: () => api.FileSystemWatcher | null) => void,
 	) => {
+		showOutput(output, OutputLevels.debug)(`Directory ${projectDir} should be watched.`);
 		addConfigWatcherIfNotExists(projectDir, () => {
-			const watcher = vscode.workspace.createFileSystemWatcher(join(projectDir, '.taq/config.json'));
-			// TODO: We should detect the event that VsCode's current Folder is changed and the watcher should be disposed
-
-			updateCommandStates(context, output, i18n, projectDir);
-
-			// TODO: Is passing these arguments to the callback of a long lived watcher prevent GC? Are these short lived objects?
-			watcher.onDidChange((e: api.Uri) => updateCommandStates(context, output, i18n, projectDir));
-			watcher.onDidCreate((e: api.Uri) => updateCommandStates(context, output, i18n, projectDir));
-			watcher.onDidDelete((e: api.Uri) => updateCommandStates(context, output, i18n, projectDir));
-			return watcher;
+			showOutput(output, OutputLevels.info)(`Adding watcher for directory ${projectDir}.`);
+			try {
+				updateCommandStates(context, output, i18n, projectDir);
+			} catch (error: any) {
+				logAllNestedErrors(error, output);
+			}
+			try {
+				const watcher = vscode.workspace.createFileSystemWatcher(join(projectDir, '.taq/config.json'));
+				// TODO: We should detect the event that VsCode's current Folder is changed and the watcher should be disposed
```

Comment:
Ali, a convention we are trying to adhere is to always append a `//TODO comment` with a link to an issue in Github. That way we can schedule this work at a later time, and and the work-to-be-done becomes more visible.

Reply:
The problem with that is, I can create many such TODOs for a simple task.
A lot of those will just be cleared as we get better insight into vscode, but making sure they are all fine can slow down initial feature implementations even further.

For instance, my current understanding is that this TODO is fine to be deleted, as when you change the folder, the vscode refreshes all the extensions. I checked that and could never see multiple watchers.

If I create issues for all such things, I can pollute the issue tracker with mostly useless issues.

If you still insist, I can do that.